src/test/common/configuration/service.vscode.test.ts

```diff
@@ -21,8 +22,15 @@ suite('Configuration Service', () => {
         expect(instanceIsSame).to.be.equal(true, 'Incorrect settings');
     });
 
-    test('Ensure async registry works', async () => {
+    test('Ensure async registry returns expected class', async () => {
         const asyncRegistry = serviceContainer.get<IAsyncDisposableRegistry>(IAsyncDisposableRegistry);
+        expect(asyncRegistry).to.be.instanceOf(AsyncDisposableRegistry);
+    });
+
+    test('Ensure async registry works', async () => {
+        // Do not retrieve AsyncDisposableRegistry from IOC container, as this will dispose all of the
```

Comment:
Split up the test, the old test was breaking the extension by disposing all singletons (a very dangerous operation).